<?xml version="1.0" encoding="utf-8"?>

<launch>
  <arg name="use_sim_time"  default="true" />
  <arg name="param_file" default="$(find nvblox_ros)/config/min_parameters.yaml"/>
  <arg name="rviz" default="false" />

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find nvblox_ros)/rviz/panda.rviz" if="$(arg rviz)"/>

  <node pkg="nvblox_ros" type="nvblox_node" name="nvblox_node" respawn="false" output="screen" >
    <rosparam file="$(arg param_file)"/>
    <remap from="depth/image" to="/wrist_camera/depth/image_rect_raw" />
    <remap from="depth/camera_info" to="/wrist_camera/depth/camera_info" />

    <remap from="color/image" to="/wrist_camera/color/image_raw" />
    <remap from="color/camera_info" to="/wrist_camera/color/camera_info" />

    <!-- Anything here ovewrites the parameters in the config file. -->
    <param name="global_frame" value="world" />
    <param name="use_sim_time" value="$(arg use_sim_time)" />
    <param name="use_lidar" value="false" />
    <param name="voxel_size" value="0.0025" />
    <param name="compute_esdf" value="false" />
    <param name="esdf_slice_height" value="0.0" />
    <param name="esdf_2d" value="false" />
    <param name="projective_integrator_max_integration_distance_m" value="0.5" />
    <param name="color_integrator_max_integration_distance_m" value="0.5" />
  </node>

</launch>
